pkg/yurthub/proxy/autonomy/autonomy.go

```diff
@@ -0,0 +1,210 @@
+		} else {
+			return node, nil
+		}
+	}
+	if node == nil {
+		return nil, fmt.Errorf("failed to get node")
+	}
+	klog.ErrorS(err, "failed to update node autonomy status")
+	return node, nil
+}
+
+func (ap *AutonomyProxy) tryUpdateNodeConditions(tryNumber int, node runtime.Object, req *http.Request) (runtime.Object, error) {
+	var originalNode, updatedNode *v1.Node
+	var err error
+	info, _ := apirequest.RequestInfoFrom(req.Context())
+
+	if node == nil {
+		if tryNumber == 0 {
+			// get from local cache
+			obj, err := ap.cacheMgr.QueryCache(req)
+			if err != nil {
+				return nil, err
+			}
+			ok := false
+			originalNode, ok = obj.(*v1.Node)
+			if !ok {
+				return nil, fmt.Errorf("could not QueryCache, node is not found")
+			}
+		} else {
+			if ap.restConfigMgr == nil {
+				return nil, fmt.Errorf("failed to initialize restConfigMgr")
+			}
+			config := ap.restConfigMgr.GetRestConfig(true)
+			client, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return nil, err
+			}
+			if tryNumber < 3 {
+				// get from apiServer cache
+				opts := metav1.GetOptions{}
+				util.FromApiserverCache(&opts)
+				originalNode, err = client.CoreV1().Nodes().Get(context.TODO(), info.Name, opts)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get node from apiserver cache")
+				}
+			} else {
+				// get from etcd
+				originalNode, err = client.CoreV1().Nodes().Get(context.TODO(), info.Name, metav1.GetOptions{})
+				if err != nil {
+					return nil, fmt.Errorf("failed to get node from apiserver cache")
+				}
+			}
+		}
+	} else {
+		originalNode = node.(*v1.Node)
+	}
+
+	if originalNode == nil {
+		return nil, fmt.Errorf("get nil node object: %s", info.Name)
+	}
+
+	changedNode, changed := ap.updateNodeConditions(originalNode)
+	if !changed {
+		return originalNode, nil
+	}
+
+	if ap.restConfigMgr == nil {
+		return originalNode, ErrRestConfigMgr
+	}
+	config := ap.restConfigMgr.GetRestConfig(true)
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return originalNode, err
+	}
+	updatedNode, err = client.CoreV1().Nodes().UpdateStatus(context.TODO(), changedNode, metav1.UpdateOptions{})
+	if err != nil {
+		return originalNode, err
+	}
+	return updatedNode, nil
+}
+
+func (ap *AutonomyProxy) updateNodeConditions(originalNode *v1.Node) (*v1.Node, bool) {
+	node := originalNode.DeepCopy()
+	if node.Annotations[projectinfo.GetAutonomyAnnotation()] == "false" || node.Labels[projectinfo.GetEdgeWorkerLabelKey()] == "false" {
+		setNodeAutonomyCondition(node, v1.ConditionFalse, "autonomy disabled", "The autonomy is disabled or this node is not edge node")
+	} else {
+		res := ap.cacheMgr.QueryCacheResult()
+		if res.ErrorKeysLength == 0 {
+			setNodeAutonomyCondition(node, v1.ConditionTrue, "autonomy enabled successfully", "The autonomy is enabled and it works fine")
+			atomic.StoreInt32(ap.cacheFailedCount, 0)
+		} else {
+			atomic.AddInt32(ap.cacheFailedCount, 1)
+			if *ap.cacheFailedCount > 3 {
```

Comment:
please optimize the code snippet as following:
```
const (
    maxCacheFailures = 3
)

currentFailures := atomic.AddInt32(ap.cacheFailedCount, 1)
if currentFailures > maxCacheFailures {
    ...
}
```

Reply:
done